File: shared/contracts/catalog_status.py
from typing import Literal

ProductStatus = Literal["draft", "published", "hidden", "deleted"]
AvailabilityStatus = Literal["available", "made_to_order", "reserved", "sold_out"]
TrendStatus = Literal["candidate", "in_review", "approved", "rejected", "published"]
CopyrightRisk = Literal["unknown", "low", "needs_review", "rejected"]

PRODUCT_STATUS: tuple[ProductStatus, ...] = ("draft", "published", "hidden", "deleted")
AVAILABILITY_STATUS: tuple[AvailabilityStatus, ...] = (
    "available",
    "made_to_order",
    "reserved",
    "sold_out",
)
TREND_STATUS: tuple[TrendStatus, ...] = (
    "candidate",
    "in_review",
    "approved",
    "rejected",
    "published",
)
COPYRIGHT_RISK: tuple[CopyrightRisk, ...] = (
    "unknown",
    "low",
    "needs_review",
    "rejected",
)

File: shared/contracts/catalog-status.ts
export const PRODUCT_STATUS = {
  DRAFT: "draft",
  PUBLISHED: "published",
  HIDDEN: "hidden",
  DELETED: "deleted"
} as const;

export type ProductStatus = (typeof PRODUCT_STATUS)[keyof typeof PRODUCT_STATUS];

export const AVAILABILITY_STATUS = {
  AVAILABLE: "available",
  MADE_TO_ORDER: "made_to_order",
  RESERVED: "reserved",
  SOLD_OUT: "sold_out"
} as const;

export type AvailabilityStatus = (typeof AVAILABILITY_STATUS)[keyof typeof AVAILABILITY_STATUS];

export const TREND_STATUS = {
  CANDIDATE: "candidate",
  IN_REVIEW: "in_review",
  APPROVED: "approved",
  REJECTED: "rejected",
  PUBLISHED: "published"
} as const;

export type TrendStatus = (typeof TREND_STATUS)[keyof typeof TREND_STATUS];

export const COPYRIGHT_RISK = {
  UNKNOWN: "unknown",
  LOW: "low",
  NEEDS_REVIEW: "needs_review",
  REJECTED: "rejected"
} as const;

export type CopyrightRisk = (typeof COPYRIGHT_RISK)[keyof typeof COPYRIGHT_RISK];
